### PR TITLE
[SQL] Add antlr4 plugin and build RikaiExt antlr4 grammar

### PR DESCRIPTION
**Pull request by eddyxu**
Introduce `sbt build` infrastructure to build Spark SQL extension.

### CODE REVIEW

File: src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4 [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    ;
+
+statement
+    : CREATE (OR REPLACE)? MODEL model=qualifiedName
+	  (OPTIONS optionList)?
+	  (AS table=qualifiedName | USING path=STRING)	# createModel
+	| (DESC | DESCRIBE) MODEL model=qualifiedName   # describeModel
+	| SHOW MODELS                                   # showModels
+	| DROP MODEL model=qualifiedName                # dropModel
+	| .*?									        # passThrough
```
Comment (bobingm) [inline src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4:29]: rm the spaces before `#`

File: src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4 [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+// Grammar for Rikai-extensions.
+grammar RikaiExtSqlBase;
+
+singleStatement
+    : statement ';'* EOF
+    ;
+
+statement
+    : CREATE (OR REPLACE)? MODEL model=qualifiedName
+	  (OPTIONS optionList)?
```
Comment (bobingm) [inline src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4]: Not related to this PR: do we want to have limited list of options?

**Review by bobingm — APPROVED** [review]

File: src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4 (hunk 2 above)
Comment (eddyxu) [reply src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4]: You meant the limited choices? I dont have strong options at the moment.  I do not feel that I can come up a good set of options now tho.

File: src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4 (hunk 1 above)
Comment (eddyxu) [reply src/main/antlr4/org/apache/spark/sql/ml/parser/RikaiExtSqlBase.g4]: Thanks. Done

**Pull request merged**